jina/executors/crafters/__init__.py

```diff
@@ -17,8 +17,11 @@ class BaseCrafter(BaseExecutor):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.required_keys = {k for k in inspect.getfullargspec(self.craft).args if k != 'self'}
+        
+    def post_init(self):
         if not self.required_keys:
             self.logger.warning(f'{self.__class__} works on keys, but no keys are specified')
+        super(BaseCrafter, self).post_init()
```

Comment:
```suggestion
        super(BaseCrafter, self).post_init()
```
but why put it into `post_init`?

Reply:
It is an undesired effect. If I want to enforce the identifier to be present from the metas when the logger is instantiated. I need to call it after the init and before the post init. 

My idea is that post init will potentially have business logic that will be good to have properly caught and handled by fluentd.

My first idea but did not see how was to inject someway eith the yaml constructor the pea logger thought

Reply:
This particular warning is to keep the same messsge as before, so no impact from the outside

Reply:
One reason we don't want to touch the `post_init` is that it is highly likely to be overridden without calling `super().post_init()`